src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -170,6 +185,16 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
    */
   private AddEventsResponse addEventsWithRetry(byte[] addEventsPayload, long startTimeMs) {
     log.debug("addEvents payload size {} bytes", addEventsPayload.length);
+
+    // 6 MB add events payload exceeded.  Log the issue and skip this message.
```

Comment:
I wonder, if we should, to be on the safe side also perform the same check on the response route - if we receive a non-200 response which indicates payload too large, we simply ignore that error and consider it as not fatal?

Reply:
Besides that, LGTM.